drivers/serial/uart_stm32.c

```diff
@@ -336,10 +336,10 @@ static int uart_stm32_configure(const struct device *dev,
 	/* Driver doesn't support 5 or 6 databits and potentially 7 or 9 */
 	if ((UART_CFG_DATA_BITS_5 == cfg->data_bits) ||
 	    (UART_CFG_DATA_BITS_6 == cfg->data_bits)
-#ifndef LL_USART_DATAWIDTH_7B
+#ifdef LL_USART_DATAWIDTH_7B
```

Comment:
This code returns `-ENOTSUP` when `LL_USART_DATAWIDTH_7B/9B` is not defined (which is the case on some parts). So, from this point of view, this code is correct.
You're modifying in "return `-ENOTSUP` when `LL_USART_DATAWIDTH_7B/9B` is defined", which doens't seem correct.
There might be things to fix, but I don't think this is the correct fix.

Reply:
You are right I havent taken mcu that dont support 9bit mode into account. I can just do a check on configuration and then return -ENOTSUP. I can have a change ready on the afternoon.

Reply:
I propose to purely remove this change.

Reply:
Tthis change was removed and requested changes have been made.